### ISSUE TITLE
Monitoring fairseq training in tensorboard does not work 

### ISSUE THREAD

**Issue by thilakshiK** [pytorch/fairseq#2374]
#### What is your question?

I am using fairseq command-line tools for training and evaluating my models.  I have enabled `--tensorboard-logdir`  to monitor in tensorboard. The `loss` and `nll_loss` shown in tensorboard and fairseq-train output is completely different. However, others are displayed correctly ( ex: `ppl, lr, etc.` ) in tensorboard. 

#### Code

**_Here is the fairseq-train output_**
   

| epoch 001:   0% 0/115 [00:00<?, ?it/s]/pytorch/torch/csrc/utils/python_arg_parser.cpp:756: UserWarning: This overload of add_ is deprecated:
	add_(Number alpha, Tensor other)
Consider using one of the following signatures instead:
	add_(Tensor other, *, Number alpha)
| epoch 001 | loss 12.337 | nll_loss 12.118 | ppl 4444.81 | wps 3123 | ups 2 | wpb 1768.696 | bsz 127.652 | num_updates 115 | lr 2.88471e-05 | gnorm 1.815 | clip 0.000 | oom 0.000 | wall 66 | train_wall 62
| epoch 001 | valid on 'valid' subset | loss 11.185 | nll_loss 10.716 | ppl 1682.5 | num_updates 115
| epoch 002 | loss 10.865 | nll_loss 10.301 | ppl 1261.85 | wps 3124 | ups 2 | wpb 1768.696 | bsz 127.652 | num_updates 230 | lr 5.75943e-05 | gnorm 1.067 | clip 0.000 | oom 0.000 | wall 134 | train_wall 123
| epoch 002 | valid on 'valid' subset | loss 10.439 | nll_loss 9.680 | ppl 820.45 | num_updates 230
| saved checkpoint checkpoints/checkpoint2.pt (epoch 2 @ 230 updates) (writing took 7.148411750793457 seconds)
| epoch 003 | loss 10.399 | nll_loss 9.640 | ppl 797.74 | wps 2952 | ups 2 | wpb 1768.696 | bsz 127.652 | num_updates 345 | lr 8.63414e-05 | gnorm 1.177 | clip 0.000 | oom 0.000 | wall 214 | train_wall 188
| epoch 003 | valid on 'valid' subset | loss 10.206 | nll_loss 9.311 | ppl 635.22 | num_updates 345 | best_loss 10.2058
| epoch 004 | loss 10.126 | nll_loss 9.269 | ppl 617.09 | wps 3122 | ups 2 | wpb 1768.696 | bsz 127.652 | num_updates 460 | lr 0.000115089 | gnorm 1.042 | clip 0.000 | oom 0.000 | wall 282 | train_wall 249
| epoch 004 | valid on 'valid' subset | loss 9.992 | nll_loss 8.999 | ppl 511.81 | num_updates 460 | best_loss 9.99156
| saved checkpoint checkpoints/checkpoint4.pt (epoch 4 @ 460 updates) (writing took 7.7255539894104 seconds)
| epoch 005 | loss 9.862 | nll_loss 8.919 | ppl 483.93 | wps 3029 | ups 2 | wpb 1768.696 | bsz 127.652 | num_updates 575 | lr 0.000143836 | gnorm 1.010 | clip 0.000 | oom 0.000 | wall 360 | train_wall 312
| epoch 005 | valid on 'valid' subset | loss 9.675 | nll_loss 8.587 | ppl 384.51 | num_updates 575 | best_loss 9.6748
| epoch 006 | loss 9.571 | nll_loss 8.528 | ppl 369.18 | wps 3106 | ups 2 | wpb 1768.696 | bsz 127.652 | num_updates 690 | lr 0.000172583 | gnorm 1.087 | clip 0.000 | oom 0.000 | wall 429 | train_wall 374
| epoch 006 | valid on 'valid' subset | loss 9.234 | nll_loss 7.996 | ppl 255.32 | num_updates 690 | best_loss 9.23367
| saved checkpoint checkpoints/checkpoint6.pt (epoch 6 @ 690 updates) (writing took 7.888668537139893 seconds)
| epoch 007 | loss 9.250 | nll_loss 8.095 | ppl 273.34 | wps 3030 | ups 2 | wpb 1768.696 | bsz 127.652 | num_updates 805 | lr 0.00020133 | gnorm 1.109 | clip 0.000 | oom 0.000 | wall 507 | train_wall 437
| epoch 007 | valid on 'valid' subset | loss 8.915 | nll_loss 7.538 | ppl 185.79 | num_updates 805 | best_loss 8.91512
| epoch 008 | loss 8.918 | nll_loss 7.650 | ppl 200.87 | wps 3110 | ups 2 | wpb 1768.696 | bsz 127.652 | num_updates 920 | lr 0.000230077 | gnorm 1.079 | clip 0.000 | oom 0.000 | wall 576 | train_wall 498
| epoch 008 | valid on 'valid' subset | loss 8.579 | nll_loss 7.056 | ppl 133.05 | num_updates 920 | best_loss 8.5792
| saved checkpoint checkpoints/checkpoint8.pt (epoch 8 @ 920 updates) (writing took 8.043779850006104 seconds)
| epoch 009 | loss 8.598 | nll_loss 7.220 | ppl 149.06 | wps 3017 | ups 2 | wpb 1768.696 | bsz 127.652 | num_updates 1035 | lr 0.000258824 | gnorm 1.102 | clip 0.000 | oom 0.000 | wall 654 | train_wall 562
| epoch 009 | valid on 'valid' subset | loss 8.283 | nll_loss 6.640 | ppl 99.7 | num_updates 1035 | best_loss 8.2829
| epoch 010 | **loss 8.292** | nll_loss 6.808 | ppl 112.08 | wps 3123 | ups 2 | wpb 1768.696 | bsz 127.652 | num_updates **1150** | lr 0.000287571 | gnorm 1.093 | clip 0.000 | oom 0.000 | wall 723 | train_wall 623
| epoch 010 | valid on 'valid' subset | **loss 7.977** | nll_loss 6.215 | ppl 74.3 | num_updates 1150 | best_loss 7.97738
| saved checkpoint checkpoints/checkpoint10.pt (epoch 10 @ 1150 updates) (writing took 8.212632417678833 seconds)
| done training in 733.6 seconds

_**Here is a screenshot of the `loss`**. I have shown the loss of train and validation at the last step (step 1150). These values do not match the output values of fairseq-train( I have bold them in output log)_ 

![Screenshot (178)](https://user-images.githubusercontent.com/35171029/88475362-a91dfb80-cf4c-11ea-8ea5-fe9d2ef44f2d.png)

#### What's your environment?
I am using **Google Colab** with TensorBoard notebook extension
 - fairseq Version: 0.9.0
 - PyTorch Version  : 1.5.1+cu101
 - OS : Windows 
 - How you installed fairseq (`pip`, source): pip install fairseq
 - Python version: 3.6
 - CUDA/cuDNN version: V10.1.243

**Comment by ghost** [comment]
Hi @thilakshiK , I encountered the same problem? Did you figure it out?
